library/std/src/io/error.rs

```diff
@@ -127,6 +133,35 @@ pub enum ErrorKind {
     /// requested to not occur.
     #[stable(feature = "rust1", since = "1.0.0")]
     WouldBlock,
+    /// A filesystem object is, unexpectedly, not a directory.
+    ///
+    /// For example, a filesystem path was specified where one of the intermediate directory
+    /// components was, in fact, a plain file.
+    #[stable(feature = "io_error_unix_more", since = "1.51.0")]
+    NotADirectory,
+    /// The filesystem object is, unexpectedly, a directory.
+    ///
+    /// A directory was specified when a non-directory was expected.
+    #[stable(feature = "io_error_unix_more", since = "1.51.0")]
+    IsADirectory,
+    /// A non-empty directory was specified where an empty directory was expected.
+    #[stable(feature = "io_error_unix_more", since = "1.51.0")]
+    DirectoryNotEmpty,
+    /// The filesystem is read-only, but a write operation was attempted.
+    #[stable(feature = "io_error_unix_more", since = "1.51.0")]
+    ReadOnlyFilesystem,
+    /// Too many levels of symbolic links.
+    ///
+    /// There was a symbolic link loop; or, the system-specific limit on the depth of symlink
+    /// traversal was exceded.
+    #[stable(feature = "io_error_unix_more", since = "1.51.0")]
+    SymbolicLinkLoop,
+    /// Stale network file handle
+    ///
+    /// With some network filesystems, notably NFS, an open file (or directory) can be invalidated
+    /// by problems with the network or server.
+    #[stable(feature = "io_error_unix_more", since = "1.51.0")]
+    StaleNetworkFileHandle,
```

Comment:
Not sure on this one. May be [`ERROR_UNRECOGNIZED_VOLUME`](https://docs.microsoft.com/en-us/windows/win32/debug/system-error-codes--1000-1299-) or [`ERROR_FILE_INVALID`](https://docs.microsoft.com/en-us/windows/win32/debug/system-error-codes--1000-1299-)?

Reply:
`ERROR_UNRECOGNIZED_VOLUME` seem to mean it didn't understand the filesystem format.  `ERROR_FILE_INVALID` probably isn't it either but I could be wrong.